crates/ntx-builder/src/builder/coordinator.rs

```diff
@@ -0,0 +1,205 @@
+    const ACTOR_CHANNEL_SIZE: usize = 100;
+
+    /// Creates a new coordinator with the specified maximum number of inflight transactions.
+    pub fn new(max_inflight_transactions: usize) -> Self {
+        Self {
+            actor_registry: HashMap::new(),
+            actor_join_set: JoinSet::new(),
+            semaphore: Arc::new(Semaphore::new(max_inflight_transactions)),
+        }
+    }
+
+    /// Spawns a new actor to manage the state of the provided network account.
+    ///
+    /// This method creates a new [`AccountActor`] instance for the specified account origin
+    /// and adds it to the coordinator's management system. The actor will be responsible for
+    /// processing transactions and managing state for accounts matching the network prefix.
+    #[tracing::instrument(name = "ntx.builder.spawn_actor", skip(self, origin, config))]
+    pub fn spawn_actor(&mut self, origin: AccountOrigin, config: &AccountActorConfig) {
+        let account_prefix = origin.prefix();
+
+        // If an actor already exists for this account prefix, something has gone wrong.
+        if let Some(handle) = self.actor_registry.remove(&account_prefix) {
+            tracing::error!("account actor already exists for prefix: {}", account_prefix);
+            handle.cancel_token.cancel();
+        }
+
+        // Construct the actor and add it to the registry for subsequent messaging.
+        let (event_tx, event_rx) = mpsc::channel(Self::ACTOR_CHANNEL_SIZE);
+        let cancel_token = tokio_util::sync::CancellationToken::new();
+        let actor = AccountActor::new(origin, config, event_rx, cancel_token.clone());
+        let handle = ActorHandle::new(event_tx, cancel_token);
+        self.actor_registry.insert(account_prefix, handle);
+
+        // Run the actor.
+        let semaphore = self.semaphore.clone();
+        self.actor_join_set.spawn(Box::pin(actor.run(semaphore)));
+
+        tracing::info!("created actor for account prefix: {}", account_prefix);
+    }
+
+    /// Broadcasts a mempool event to all active account actors.
+    ///
+    /// This method distributes the provided event to every actor currently registered
+    /// with the coordinator. Each actor will receive the event through its dedicated
+    /// message channel and can process it accordingly.
+    ///
+    /// If an actor fails to receive the event, it will be canceled.
+    pub async fn broadcast_event(&mut self, event: &MempoolEvent) {
+        tracing::debug!(
+            actor_count = self.actor_registry.len(),
+            "broadcasting event to all actors"
+        );
+
+        let mut failed_actors = Vec::new();
+
+        // Send event to all actors.
+        for (account_prefix, handle) in &self.actor_registry {
+            if let Err(err) = Self::send(handle, event).await {
+                tracing::error!("failed to send event to actor {}: {}", account_prefix, err);
+                failed_actors.push(*account_prefix);
+            }
+        }
```

Comment:
This is probably fine for now, but I'm worried that this won't scale in the future. For example, if we have 10M actors, we'd be notifying all of them with every single transaction etc. This may create a very significant overhead.

Maybe it would be better if the coordinator could do a bit more work in figuring out which actors are interested in specific events and forward these events only to relevant actors. This shouldn't be too difficult (I think), but we'll need to think through how to do this cleanly and efficiently.